apps/server/src/services/auth0Service.test.ts

```diff
@@ -0,0 +1,94 @@
+import { http, HttpResponse } from 'msw'
+import config from '../config'
+import * as logger from '../infra/logger'
+import { server } from '../test/testServer'
+import { getAccessToken } from './auth0Service'
+import * as redisClient from './storageClient/redis'
+
+describe('getAccessToken', () => {
+  it('should return cached access token if available', async () => {
+    const cachedToken = 'cached-access-token'
+
+    jest.spyOn(redisClient, 'get').mockResolvedValue(cachedToken)
+
+    const result = await getAccessToken()
+
+    expect(result).toBe(cachedToken)
+  })
+
+  it('should fetch a new token and cache it if not cached', async () => {
+    const mockAccessToken = 'new-access-token'
+    const mockResponseJson = {
+      access_token: mockAccessToken,
+      expires_in: 3600
+    }
+
+    let createAccessTokenPayload: any = null
+
+    server.use(
+      http.post(config.Auth0TokenUrl, async ({ request }) => {
+        createAccessTokenPayload = await request.json()
+
+        return HttpResponse.json(mockResponseJson)
+      })
+    )
+
+    jest.spyOn(redisClient, 'get').mockResolvedValue(null)
+    const setSpy = jest.spyOn(redisClient, 'set').mockResolvedValue(undefined)
+
+    const token = await getAccessToken()
+
+    expect(createAccessTokenPayload).toEqual({
+      grant_type: 'client_credentials',
+      audience: 'ucp-widget-interactions',
+      client_id: config.UCPClientId,
+      client_secret: config.UCPClientSecret
+    })
+
+    expect(setSpy).toHaveBeenCalledWith('auth0AccessToken', mockAccessToken, {
+      EX: 3600
+    })
+
+    expect(token).toBe(mockAccessToken)
+  })
+
+  it('should log a warning and return null when unauthorized', async () => {
+    jest.spyOn(redisClient, 'get').mockResolvedValue(null)
+    const setSpy = jest.spyOn(redisClient, 'set')
+    const warningLogSpy = jest.spyOn(logger, 'warning')
+
+    server.use(
+      http.post(config.Auth0TokenUrl, async ({ request }) => {
+        return new HttpResponse(null, {
+          status: 401,
+          statusText: 'Unauthorized'
+        })
+      })
+    )
+
+    const token = await getAccessToken()
+
+    expect(token).toBeNull()
+    expect(setSpy).not.toHaveBeenCalled()
+    expect(warningLogSpy).toHaveBeenCalledWith(
+      'Unauthorized to retrieve UCP access token'
+    )
+  })
+
+  it('should handle fetch errors gracefully', async () => {
+    jest.spyOn(redisClient, 'get').mockResolvedValue(null)
+    const setSpy = jest.spyOn(redisClient, 'set')
+
+    server.use(
+      http.post(config.Auth0TokenUrl, async ({ request }) => {
+        return HttpResponse.error()
+      })
+    )
+
+    await expect(getAccessToken()).rejects.toThrow(
```

Comment:
Does this still log if you returned with a status code of 500 instead of HttpResponse.error()?